server/bin/test/test_pbench_server_put_shim_001.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import pytest
+
+from pathlib import Path
+
+from pbench_server_put_shim_001 import get_receive_dir, process_tb, Results
+from pbench.common.logger import get_pbench_logger
+from pbench.common.results import CopyResultTb
+from pbench.test.unit.server.common import tarball, bad_tarball
+
+
+class FileUploadError(Exception):
+    pass
+
+
+class TestProcessTb:
+    @pytest.fixture(autouse=True)
+    def config_and_logger(self, valid_config):
+        # Setup the configuration and logger
+        self.config = valid_config
+        self.logger = get_pbench_logger("pbench", valid_config)
+        self.log_file = self.config.LOGSDIR / "pbench" / "pbench.log"
+        self.tbname = Path(tarball).name
+
+        yield
+        # Teardown the setup
+        self.config, self.logger = None, None
+
+    def get_last_log_line(self):
+        logFile = open(self.log_file)
+        lines = list(logFile)
+        return lines[-1].replace("\n", "")
```

Comment:
You're assuming that the log entry you want must be the last one written; that's easily confused by something like an `INFO` message someone throws in to debug. Using `caplog` and filtering out `ERROR` log entries (regardless of their position in the log) is much more reliable.

Reply:
yeah right. I will just use caplog to determine